### ISSUE TITLE
Setting up a proxy setup with self signed certificates doesn't work.

### ISSUE THREAD

**Issue by dpakach** [owncloud/ocis#301]
Setting up a proxy setup with self signed certificates doesn't work.

### steps to reproduce
run ocis with ldap user driver, self signed certificates and these settings.
```
export REVA_STORAGE_OC_DATA_SERVER_URL='https://172.17.0.1:9200/data'
export OCIS_CONFIG_FILE=config/proxy-example.json
export PROXY_TRANSPORT_TLS_CERT=ocisWithProxy/selfsigned.crt
export PROXY_TRANSPORT_TLS_KEY=ocisWithProxy/selfsigned.key
```
Here trying to set data server url to `172.17.0.1` in order to run phoenix from different host (for eg. from within a docker container while ocis is running in host machine)
And the selfsigned certificates are valid for both `localhost` and `172.17.0.1`

add extra entry in proxy config for the `/data` endpoint
```
{
  "endpoint": "/data/",
  "backend": "http://localhost:9164"
},
```
with ocis running upload a file and try to access it using a dav endpoint from different host than localhost

```
curl https://172.17.0.1:9200/remote.php/dav/files/tu1/new-lorem.txt -u tu1:1234 --insecure -v
```

### Result 
we get 500 Internal Server Error in response

The logs in server show
```
2020-06-02T18:16:27+05:45 ERR error performing http request error="Get https://172.17.0.1:9200/data/tu1/new-lorem.txt: x509: certificate signed by unknown authority" pkg=rhttp service=reva traceid=02a0946d1924d7f48a70898cf10f039a
2020-06-02T18:16:27+05:45 ERR http end="02/Jun/2020:18:16:27 +0545" host=127.0.0.1 method=GET pkg=rhttp proto=HTTP/1.1 service=reva size=0 start="02/Jun/2020:18:16:27 +0545" status=500 time_ns=23970313 traceid=02a0946d1924d7f48a70898cf10f039a uri=/remote.php/dav/files/tu1/new-lorem.txt url=/remote.php/dav/files/tu1/new-lorem.txt

```

## Conclusion
This may be the expected behavior from the security standpoint but for testing and other purposes it would be nice to just run it with self signed certificates by supressing the self-signed certificate errors.

Also there may be some problems in my configuration or this may not be the way to do things correctly.
cc @butonic @IljaN

**Comment by dpakach** [comment]
Blocking 
https://github.com/owncloud/phoenix/pull/3554
https://github.com/owncloud/phoenix/pull/3488

**Comment by wkloucek** [comment]
@dpakach seems to be fixed? See #3455 especially https://github.com/owncloud/web/issues/3455#issuecomment-730875086
